Hexastore/Processor/StoreOperationFactory.cs

```diff
@@ -3,46 +3,37 @@
 using System.Collections.Generic;
 using System.Text;
 using System.Threading;
+using System.Threading.Tasks;
 
 namespace Hexastore.Processor
 {
     public class StoreOperationFactory : IStoreOperationFactory
     {
-        private readonly ConcurrentDictionary<string, ReaderWriterLock> _locks = new ConcurrentDictionary<string, ReaderWriterLock>();
-
-        public IDisposable Read(string storeId)
-        {
-            var storeLock = _locks.GetOrAdd(storeId, (x) =>
-            {
-                return new ReaderWriterLock();
-            });
-            storeLock.AcquireReaderLock(10_000);
-            return new StoreOperation(storeLock);
-        }
+        private readonly ConcurrentDictionary<string, Mutex> _locks = new ConcurrentDictionary<string, Mutex>();
 
         public IDisposable Write(string storeId)
         {
             var storeLock = _locks.GetOrAdd(storeId, (x) =>
             {
-                return new ReaderWriterLock();
+                return new Mutex();
             });
-            storeLock.AcquireWriterLock(10_000);
+            storeLock.WaitOne();
```

Comment:
Why the switch to a Mutex here? To my knowledge, they are slower than ReaderWriterLock(slim)'s and more computationally expensive. I know they support cross process, but we don't need locking across processes, do we? Also, why the switch to allow WaitOne() to block indefinitely without a timeout?